UIComponents/Sources/UIComponents/Components/Transitions/ZoomTransition/Animations/TransitionAnimator.swift

```diff
@@ -0,0 +1,46 @@
+import UIKit
+
+class TransitionAnimator: NSObject, UIViewControllerAnimatedTransitioning {
+    var toViewController: UIViewController!
+    var fromViewController: UIViewController!
+    let inDuration: TimeInterval
+    let outDuration: TimeInterval
+    let direction: AnimationDirection
+
+    init(inDuration: TimeInterval, outDuration: TimeInterval, direction: AnimationDirection) {
+        self.inDuration = inDuration
+        self.outDuration = outDuration
+        self.direction = direction
+        super.init()
+    }
+
+    internal func transitionDuration(using transitionContext: UIViewControllerContextTransitioning?) -> TimeInterval {
+        return direction == .in ? inDuration : outDuration
+    }
+
+    internal func animateTransition(using transitionContext: UIViewControllerContextTransitioning) {
```

Comment:
please take a look at
let finished = !transitionContext.transitionWasCancelled
transitionContext.completeTransition(finished)

and 
guard let sourceView = transitionContext.view(forKey: .from)
        else {
            transitionContext.completeTransition(false)
            return
        }

        guard transitionContext.isAnimated
        else {
            transitionContext.completeTransition(true)
            return
        }
        
maybe we need them